src/multiselect_controls.js

```diff
@@ -0,0 +1,418 @@
+ */
+
+
+/**
+ * @fileoverview Multiple selection svg switch.
+ */
+
+import * as Blockly from 'blockly/core';
+
+import DragSelect from '../lib/ds.min';
+import {blockSelection, inMultipleSelectionMode,
+  setSelectionMode} from './global';
+
+/**
+ * Icon path of the multi select controls when enabled.
+ * @type {string}
+ */
+let ENABLED_IMG = 'https://github.com/mit-cml/workspace-multiselect/raw/main/test/media/select.svg';
+
+/**
+ * Icon path of the multi select controls when disabled.
+ * @type {string}
+ */
+let DISABLED_IMG = 'https://github.com/mit-cml/workspace-multiselect/raw/main/test/media/unselect.svg';
+
+/**
+ * Width of the multi select controls.
+ * @type {number}
+ * @const
+ */
+const WIDTH = 32;
+
+/**
+ * Height of the multi select control.
+ * @type {number}
+ * @const
+ */
+const HEIGHT = 32;
+
+/**
+ * Distance between multi select controls and bottom or top edge of workspace.
+ * @type {number}
+ * @const
+ */
+const MARGIN_VERTICAL = 20;
+
+/**
+ * Distance between multi select controls and right or left edge of workspace.
+ * @type {number}
+ */
+const MARGIN_HORIZONTAL = 20;
+
+/**
+ * Class for a multi select controls.
+ * @implements {Blockly.IPositionable}
+ */
+export class MultiselectControls {
+  /**
+   * @param {!Blockly.WorkspaceSvg} workspace The workspace to sit in.
+   * @param {Object} options The icons configuration.
+   */
+  constructor(workspace, options) {
+    /**
+     * @type {!Blockly.WorkspaceSvg} The workspace to sit in.
+     * @private
+     */
+    this.workspace_ = workspace;
+
+    /**
+     * State of the multi select controls.
+     * @type {boolean}
+     */
+    this.enabled = false;
+
+    /**
+     * The unique id for this component that is used to register with the
+     * ComponentManager.
+     * @type {string}
+     */
+    this.id = 'multiselectControls';
+
+    /**
+     * A handle to use to unbind the mouse down event handler for multi select
+     *    button. Opaque data returned from browserEvents.conditionalBind.
+     * @type {?Blockly.browserEvents.Data}
+     * @private
+     */
+    this.onMultiselectWrapper_ = null;
+
+    /**
+     * The multi select svg <g> element.
+     * @type {SVGGElement}
+     * @private
+     */
+    this.multiselectGroup_ = null;
+
+    /**
+     * Whether this has been initialized.
+     * @type {boolean}
+     * @private
+     */
+    this.initialized_ = false;
+
+    /**
+     * Store the status of workspace dragging
+     * before entering multi select mode.
+     * @type {boolean}
+     * @private
+     */
+    this.hasDisableWorkspaceDrag_ = false;
+
+    /**
+     * Store the block that just got unselected.
+     * @type {!Blockly.BlockSvg}
+     * @private
+     */
+    this.justUnselectedBlock_ = null;
+
+    if (options && options.enabledIcon) {
+      ENABLED_IMG = options.enabledIcon;
+    }
+
+    if (options && options.disabledIcon) {
+      DISABLED_IMG = options.disabledIcon;
+    }
+  }
+  /**
```

Comment:
Small code style issue: I'd like to have a blank line between the end of a function decl and before the jsdoc comment for the next function. Apply this to the remainder of the file.